pybop/problems/fitting_problem.py

```diff
@@ -134,3 +135,112 @@ def evaluateS1(self, inputs: Inputs):
         )
 
         return (y, np.asarray(dy))
+
+
+class MultiFittingProblem(BaseProblem):
+    """
+    Problem class for joining mulitple fitting problems.
+
+    Extends `FittingProblem` to multiple fitting problems.
+    """
+
+    def __init__(self, problem_list, weights=None):
+        self.problem_list = problem_list
+        self.weights = weights
+
+        # Compile the set of parameters, ignoring duplicates
+        combined_parameters = Parameters()
+        for problem in self.problem_list:
+            combined_parameters.join(problem.parameters)
+
+        # Combine the target datasets
+        combined_dataset = Dataset(
+            {"Time [s]": np.asarray([]), "Combined signal": np.asarray([])}
+        )
+        for problem in self.problem_list:
+            for signal in problem.signal:
+                combined_dataset["Time [s]"] = np.concatenate(
+                    (combined_dataset["Time [s]"], problem._time_data)
+                )
+                combined_dataset["Combined signal"] = np.concatenate(
+                    (combined_dataset["Combined signal"], problem._target[signal])
+                )
```

Comment:
I think these objects can be formed as a list then concatenated once at the end. `np.concatenate` is on the expensive side, and completing it once on a larger object can save overhead.